### ISSUE TITLE
Does 4.1.3 Status Messages mean every status message needs to have a role="status"/"alert"?

### ISSUE THREAD

**Issue by yatil** [w3c/wcag#1475]
In a discussion in [The Accessibility Slack ↗](https://web-a11y.slack.com/archives/C042TSFGN/p1602171734008200)™ there is a question on [4.1.3 Status Messages ↗](https://www.w3.org/TR/WCAG21/#status-messages). The core of question is (as far as I have understood it):

If there are multiple errors in a form field, and on click of the submit button multiple error messages show up (individual messages for individual fields), does 4.1.3 Status Messages *require* the use of `role="status"` or `role="alert"` on every single error message?

Would it be sufficient to have a “main” error message (“there are errors on the form”) and using another way to convey the particular error (for example setting the focus into the first erroneous field and using `aria-describedby` to reference the particular error)?

Or, as 4.1.3 Status Messages says “[…]status messages can be programmatically determined through role **or properties** such that they can be presented to the user by assistive technologies without receiving focus”, is the `aria-describedby` (alternatively adding a `aria-label`) sufficient as the focus is set into the field (not to the error message) and therefore the error message is triggered without being focused itself?

The concern here is that, if all status messages need to be announced using a `role` or a live region, it might be a lot of noise in assistive technologies.

(Note: As GitHub is still distinguishing links only by color, I added a ↗ symbol to indicate links.)

**Comment by smhigley** [comment]
Just a bit of clarification -- the original question was whether a form error message would need to be a live region when focus is not handled by the author, for example messages that appear on blur, or that appear dynamically as a user types. The proposal was to update `aria-describedby` to create a programmatic association between the field and the error message, but without using a live region or author-imposed focus change.

The complication that was raised is that if every error message in a form is a live region in order to handle errors on input/blur, then displaying all errors on submit would trigger multiple status messages at once, which seems undesirable. There are ways an author could get around this (e.g. remove the live region roles or attributes for errors triggered through a submit event), though most authors would probably not take the trouble of doing that.

I personally don't see how `aria-describedby` could be enough to programmatically identify something as a status message, though I do agree that there are some cases where content appearing on the screen may technically fit the definition of "status message," but that might be subjectively judged to not be important enough to warrant a status message in a specific user context.

A better example might be: a form does not move focus on submit, but displays one for each invalid form field. Technically all of the visible field errors would meet the definition if "status message," so one could interpret the criterion to mean that they all need to be marked up as status messages. But the best UX would more likely be to have one single status message for the entire form (or a change of focus).

**Comment by mraccess77** [comment]
Personally I'd say if focus is moved to the field with associated error or message then a live region is not needed.  However, if error message appears when the control loses focus then the message does need a live region because the describedby association is not enough.   If you had one single area that updated with errors then assigning the status role to that area seems like it would be sufficient as we have aria-relevant to allow for control of what is announced.

**Comment by yatil** [comment]
This has been open for three years and not been discussed for three years, so I’ll close the issue.

**Comment by patrickhlauke** [comment]
Even though it's closed now (which alerted me to it just now for the first time), I'll just add: the point of the SC, to me, is to make sure that somebody who can't see the whole of the screen still gets the same/similar information outcome as somebody who can. So it's not a mechanistic "make sure every thing that updates has a particular `role` or something", but more abstractly about making sure SR users get the same info (which could be done with a single announcement and then sub-announcements as separate fields are then focused, etc). i.e. in the end, an SR user is not disadvantaged for not being able to notice visual changes/statuses. at least, that's my understanding/interpretation (and no doubt there will be others who interpret it differently)

**Comment by paulbohman** [comment]
Even though this issue was just closed, I submitted a related issue https://github.com/w3c/wcag/issues/3543 because there are some use cases in which making a status announcement in real time is bad for screen reader usability, such as when announcing an error when a user leaves a form field. The result is that the screen reader first reads the new item that just received the focus, then announces the error message for the previous item. It sounds as if the error is for the new item, but it is not.
